### DISCUSSION TITLE
Memory leak in v10 with Image component?

### DISCUSSION THREAD

**Discussion by josebrito** [vercel/next.js#19597] [Help] [answered]
Hi,

After the first release of our site in production with Next@v10 (v10.0.2 precisely), we noticed a gradual memory consumption pattern in our servers.

**Edit:** worth noting that we've skipped 9.5 and upgraded straight from 9.4. So it could be a 9.5 issue.

**Edit:** **maybe not related to Image component**, as we stopped using it, and still notice a gradual increase in memory consumption (which didn't happen before v10). See [this comment](#discussioncomment-144661).

**Edit: Jan 8th 2021:** this is definitely related to Image component. We gave it a second chance, but had to rollback due to very high memory usage.

The only updated library in these releases was Next, and we do use the new Image component in a very limited set of images:

- I estimate that no more than 50 images are being optimised with the Image component
- All of them are above the fold (hero images, with a Max Age of 1 hour defined in the Cache Control header at the original source)
- Our custom config in `next.config.js`:

```
  images: {
    domains: ['storage.googleapis.com'],
    deviceSizes: [500, 575, 750, 1150, 1500, 1920],
  },
```

Has anyone experienced this as well? Could it have any relation with the Image component? If so, it is worrying, as we are using it only for a very limited set of images, and we have plans to adopt it for a much larger set of images (e.g. product images).

As you can see in the image below, up until Next v10 the memory consumption was pretty steady. Let me know if we can provide some more details.

![Next 10 releases](https://i.imgur.com/CKTExcr.png)

**Comment by josebrito** [comment]
**UPDATE**

We have deployed a new release **without** Image component usages, and observe the same pattern.

![Next 10 releases](https://i.imgur.com/icCexm2.png)

**Comment by danielxfriberg** [comment]
I updated a clients website to nextJS 10.0.3 a day after release (12 days ago). Last week I updated all images to the new Image component and have since then been receiving memory warnings from Heroku, where the client is hosted.

This image is from Heroku, where you can cleary see where I updated to the new Image component.
![image](https://user-images.githubusercontent.com/70016792/101357083-42650300-3899-11eb-95e6-c9ea713f41df.png)

**Comment by keiraarts** [comment]
I can verify the same thing. Every new unique Next/Image component increases the JS heap size and it doesn't seem to ever be garbage collected. This is on a Next Koa production server.

**Comment by CRAKZOR** [comment]
guess when we switched to v10 image components|
![Screen Shot 2020-12-17 at 5 44 59 PM](https://user-images.githubusercontent.com/50730412/102556992-d1d79680-408f-11eb-93e3-eba057aac3c1.png)

**Comment by lukas-bryla** [comment]
I can confirm this, I went from:
    "next": "9.5.3",
to:
    "next": "^10.0.5",

implemented the Image components and suddenly my server is bloated. Nothing else was committed at the time when memory went from like 10% to 170%. I will now try to take next/Image off my website and observe if it helps.

**Comment by josebrito** [comment, marked as answer]
Will close this discussion as it's already under investigation here: https://github.com/vercel/next.js/issues/20915

If anyone needs to give additional details please do so in the issue thread.
